### PR TITLE
add task 2 edited

### PR DESCRIPTION
**Pull request by InnaSamarina**

### CODE REVIEW

File: Task2_2.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+for k, v in tmp_dict.items():
+    if v["count"] == 1:
+        result_key = k
+    else:
+        result_key = "%s_%d" % (k, v["index"])
+
+    result_dict[result_key] = v["max"]
+
+print('Result dict:', result_dict)
+
```
Comment (arastopchin) [inline Task2_2.py:38]: image
I run yor program on small dict size, seems it doesn't work properly :(

File: Task2_2.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import string
+import random
+
+# creating list of dicts
+list_of_dicts = []
+# set number of dicts
+for k in range(random.randint(2, 10)):
+    # add random keys and values and number of key-value pairs
+    mydict = {chr(random.randint(ord('a'), ord('z'))): random.randint(0, 100) for i in range(random.randint(1, 100))}
```
Comment (arastopchin) [inline Task2_2.py:9]:
```suggestion
    mydict = {chr(random.randint(ord('a'), ord('z'))): random.randint(0, 100) for _ in range(random.randint(1, 100))}
```
Usually if you use for loop just to iterate through and don't use iterable value it's omitted by underscore.

File: Task2_2.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import string
```
Comment (arastopchin) [inline Task2_2.py:1]: Do you use this module somewhere?

File: Task2_2.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import random
+
+# creating list of dicts
+list_of_dicts = []
+# set number of dicts
+for k in range(random.randint(2, 10)):
```
Comment (arastopchin) [inline Task2_2.py:7]: 1) You don't use 'k' anywhere inside loop, then omit it by _
2) Variable names should be **meaningful**, please no more i, j, k etc.

File: Task2_2.py (hunk 2 above)
Comment (arastopchin) [reply Task2_2.py:9]: Moreover no mydict, yourdict, finaldict, prefinaldict etc.

File: Task2_2.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    list_of_dicts.append(mydict)
+print('list of dicts:', list_of_dicts)
+
+#creating temporary dict
+tmp_dict = {}
+#getting numbers of list elements
+for i, d in enumerate(list_of_dicts):
+    #iterating each dict in list
+    for k, v in d.items():
+        if k not in tmp_dict:
```
Comment (arastopchin) [inline Task2_2.py:20]: As you can see here using such variables as i, d, k, v complicate readability of your code for others, it's really hard to catch already what is what here

Comment (arastopchin) [reply Task2_2.py:20]:
```suggestion
for index, dict in enumerate(list_of_dicts):
    #iterating each dict in list
    for letter, value in dict.items():
        if letter not in tmp_dict:
```

**Review by arastopchin — APPROVED** [review]
It's approved by now as we don't have time for corrections, but there is future homework where you will need to do the same but using functions. You'll need to correct your logic to program work as expected.

File: Task2_2.py (hunk 2 above)
Comment (InnaSamarina) [reply Task2_2.py:9]: Got it. But why can't I use names like finaldict, resultdict and others? They explain what is contained in the variable.

File: Task2_2.py (hunk 3 above)
Comment (InnaSamarina) [reply Task2_2.py:1]: I was going to use, but I didn't

File: Task2_2.py (hunk 4 above)
Comment (InnaSamarina) [reply Task2_2.py:7]: Got it.

File: Task2_2.py (hunk 2 above)
Comment (arastopchin) [reply Task2_2.py:9]: Actually no, they don't :) What sense should I see in finaldict? :)

File: Task2_2.py (hunk 4 above)
Comment (arastopchin) [reply Task2_2.py:7]: Please remove all unneccassary imports next time

Comment (InnaSamarina) [reply Task2_2.py:7]: Ok!

File: Task2_2.py (hunk 2 above)
Comment (InnaSamarina) [reply Task2_2.py:9]: Well, the title says this version of the dictionary is final. But OK, I'll write something more meaningful.

Comment (arastopchin) [reply Task2_2.py:9]: https://builtin.com/data-science/variable-names

Quite comprehensive tutorial :)

**Pull request merged**